aten/src/ATen/native/mps/operations/UnaryOps.mm

```diff
@@ -12,15 +12,20 @@
 namespace mps {
 
 typedef MPSGraphTensor* (^UnaryOpBlock)(MPSGraph*, MPSGraphTensor*);
+using is_noop_p = std::function<bool(const Tensor&)>;
 
-void unary_op(const Tensor& self, const Tensor& output, std::string op_name, UnaryOpBlock unaryBlock)
+
+bool is_empty_tensor(const Tensor& self) {
+  return self.numel() == 0;
+}
+
+void unary_op(const Tensor& self, const Tensor& output, std::string op_name, UnaryOpBlock unaryBlock, is_noop_p is_noop = is_empty_tensor)
```

Comment:
Hi,
May i know what happen if the unary ops get the input with double precision? I see the binary ops will have assertion:https://github.com/pytorch/pytorch/blob/71d020262793542974cf13b30f2a9099773f015c/aten/src/ATen/native/mps/operations/BinaryKernel.mm#L291